prism-agent/service/server/src/main/scala/io/iohk/atala/agent/server/jobs/BackgroundJobs.scala

```diff
@@ -1001,6 +1107,35 @@ object BackgroundJobs {
     } yield agent
   }
 
+  private[this] def handleCredentialErrors
+      : PartialFunction[Throwable | CredentialServiceError, CredentialServiceError] = {
+    case e: CredentialServiceError => e
+    case t: Throwable              => CredentialServiceError.UnexpectedError(t.getMessage())
+  }
+
+  private[this] def handlePresentationErrors: PartialFunction[
+    Throwable | CredentialServiceError | PresentationError | BackgroundJobError | List[String],
+    PresentationError | CredentialServiceError | BackgroundJobError | List[String]
+  ] = {
+    case c: CredentialServiceError => c
+    case p: PresentationError      => p
+    case b: BackgroundJobError     => b
+    case x: List[String]           => x
```

Comment:
This pattern cannot be checked at runtime. It's coming from `JwtPresentation.verify`, maybe we want to transform it earlier.